src/custom-resource-loader.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+// eslint-disable-next-line import/no-unassigned-import
+import "./patch-promise.js";
+
+import {ResourceLoader} from "jsdom";
+
+import logging from "./logging.js";
+import fetchPackage from "./fetch_package.js";
+
+import type {FetchOptions} from "jsdom";
+import type {RequestStats} from "./types.js";
+
+export class CustomResourceLoader extends ResourceLoader {
+    _active: boolean;
+    _requestStats: ?RequestStats;
+
+    /**
+     * We will return EMPTY in cases where we just don't care about the file
+     * loading. Let's just reuse a promise for that.
+     */
+    static EMPTY = Promise.resolve(Buffer.from(""));
+
+    constructor(requestStats?: RequestStats) {
+        super();
+
+        this._active = true;
+        this._requestStats = requestStats;
+    }
+
+    get isActive(): boolean {
+        return this._active;
+    }
```

Comment:
This is new (I don't like tests delving into internals, so I added this so we can confirm that construction and `close` do what they are supposed to).